src/instructlab/model/dk_bench_utils.py

```diff
@@ -0,0 +1,320 @@
+
+logger = logging.getLogger(__name__)
+
+
+class IOFileType(enum.Enum):
+    CSV = "csv"
+    JSONL = "jsonl"
+    XLSX = "xlsx"
+
+
+def validate_output_file_formats(file_formats: List[str]):
+    for file_format in file_formats:
+        valid_formats = {item.value for item in IOFileType}
+        if file_format not in valid_formats:
+            raise ValueError(
+                f"File format {file_format} is not a valid output format. Format must be one of {valid_formats}"
+            )
+
+
+def create_results_file_name(
+    file_format: str, output_dir: str, timestamp: str, model_name: str
+) -> str:
+    if IOFileType.CSV.value == file_format:
+        file_type = "csv"
+    elif IOFileType.XLSX.value == file_format:
+        file_type = "xlsx"
+    elif IOFileType.JSONL.value == file_format:
+        file_type = "jsonl"
+    else:
+        raise ValueError("File format is not one of: csv, xlsx, jsonl")
+
+    output_path = pathlib.Path(output_dir)
+    if not output_path.exists() or not output_path.is_dir():
+        output_dir = DEFAULTS.EVAL_DATA_DIR
+        logger.debug(
+            "Output dir provided is not a directory or does not exist, writing results to default evaluation output directory %s",
+            output_dir,
+        )
+
+    return f"{output_dir}/responses-{model_name}-{timestamp}.{file_type}"
+
+
+def make_run_dir(output_dir: str) -> str:
+    now = datetime.now()
+    timestamp = now.isoformat()
+    # remove any training slashes
+    output_dir = os.path.normpath(output_dir)
+    run_dir = f"{output_dir}/job-{timestamp}"
+
+    path_does_not_exist = not os.path.exists(output_dir)
+    path_is_not_dir = not os.path.isdir(output_dir)
+    if path_does_not_exist or path_is_not_dir:
+        run_dir = f"{DEFAULTS.EVAL_DATA_DIR}/job-{timestamp}"
+        logger.debug(
+            "User provided output directory path %s does not exist or is not a directory. Creating directory for results in %s",
+            output_dir,
+            DEFAULTS.EVAL_DATA_DIR,
+        )
+
+    os.makedirs(run_dir, exist_ok=True)
+    logger.debug("Created directory for results at %s", run_dir)
+    return run_dir
+
+
+def print_results(result: EvaluationResult, results_files: List[str], model_name: str):
+    print("\n")
+    print("# DK-BENCH REPORT")
+    print(f"\n## MODEL: {model_name}\n")
+    total_score = 0
+    for i, score in enumerate(result.scores):
+        print(f"Question #{i+1}:     {score['domain_specific_rubrics']}/5")
+        total_score += score["domain_specific_rubrics"]
+
+    average = total_score / len(result.scores)
+    print("----------------------------")
+    print(f"Average Score:   {average:.2f}/5")
+    print(f"Total Score:     {total_score}/{len(result.scores)*5}\n")
+
+    print("Responses and scores written to:")
+    for file in results_files:
+        print(f"{file}")
+
+
+def create_excel_results_file(excel_file: str, result: EvaluationResult):
+    summary_df = pd.DataFrame()
```

Comment:
this is a function that should have a docstring, as the logic here can be quickly broken by someone who does not understand what is going on